syntax/blocks.js

```diff
@@ -239,28 +243,45 @@ var english = {
 
   dropdowns: {},
 
-  commands: {},
+  commands: [],
 }
 allBlocks.forEach(function(info) {
-  english.commands[info.spec] = info.spec
+  english.commands[info.id] = info.spec
 })
 loadLanguages({
   en: english,
 })
 
 /*****************************************************************************/
 
+function isBlockSupported(block, version) {
+  switch (version) {
+    case "scratch2":
+      return block.selector
+    case "scratch3":
+      return true // Scratch 3.0 supports all 2.0 blocks
+    default:
+      return true
+  }
+}
+
+function registerCheck(id, func) {
+  if (!blocksById[id]) throw new Error("Unknown ID: " + id)
+  blocksById[id].accepts = func
+}
+
 function specialCase(id, func) {
   if (!blocksById[id]) throw new Error("Unknown ID: " + id)
   blocksById[id].specialCase = func
 }
 
 function disambig(id1, id2, test) {
-  var func = function(info, children, lang) {
-    return blocksById[test(children, lang) ? id1 : id2]
-  }
-  specialCase(id1, func)
-  specialCase(id2, func)
+  registerCheck(id1, function(info, children, lang) {
+    return test(children, lang)
+  })
+  registerCheck(id2, function(info, children, lang) {
+    return !test(children, lang)
```

Comment:
Do we need this, or can we rely on the fact that the first ID won’t pass the `accepts()` check?

Reply:
This is just current behavior that has been rewritten. Maybe we can change. Or maybe not.